File: populate-service/src/model/Paciente.java
package model;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import service.IGenericData;

public class Paciente implements IGenericData {

  private String id;
  private String nome;
  private String cpf;
  private LocalDateTime dataNascimento;
  private String sexo;
  private String telefone;
  private String email;
  private String endereco;
  private String senha;
  private String RGM;
  private List<Exame> exames = new ArrayList<>();

  // Getters e Setters

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getNome() {
    return nome;
  }

  public void setNome(String nome) {
    this.nome = nome;
  }

  public String getCpf() {
    return cpf;
  }

  public void setCpf(String cpf) {
    this.cpf = cpf;
  }

  public LocalDateTime getDataNascimento() {
    return dataNascimento;
  }

  public void setDataNascimento(LocalDateTime dataNascimento) {
    this.dataNascimento = dataNascimento;
  }

  public String getSexo() {
    return sexo;
  }

  public void setSexo(String sexo) {
    this.sexo = sexo;
  }

  public String getTelefone() {
    return telefone;
  }

  public void setTelefone(String telefone) {
    this.telefone = telefone;
  }

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public String getEndereco() {
    return endereco;
  }

  public void setEndereco(String endereco) {
    this.endereco = endereco;
  }

  public String getSenha() {
    return senha;
  }

  public void setSenha(String senha) {
    this.senha = senha;
  }

  public String getRGM() {
    return RGM;
  }

  public void setRGM(String RGM) {
    this.RGM = RGM;
  }

  public List<Exame> getExames() {
    return exames;
  }

  public void setExames(List<Exame> exames) {
    this.exames = exames;
  }

  public void addExame(Exame exame) {
    this.exames.add(exame);
  }
}

File: webapp/main/php/model/Paciente.php
<?php
require_once __DIR__ . '/../dao/IGenericMainData.php';
class Paciente implements IGenericMainData
{
    private string $id;
    private string $nome;
    private string $cpf;
    private DateTime $dataNascimento;
    private string $sexo;
    private string $telefone;
    private string $email;
    private string $endereco;
    private ?string $RGM;
    private array $exames = []; // Lista de objetos Exame

    // Getters e Setters

    public function getId(): string
    {
        return $this->id;
    }

    public function setId(string $id): void
    {
        $this->id = $id;
    }

    public function getNome(): string
    {
        return $this->nome;
    }

    public function setNome(string $nome): void
    {
        $this->nome = $nome;
    }

    public function getCpf(): string
    {
        return $this->cpf;
    }

    public function setCpf(string $cpf): void
    {
        $this->cpf = $cpf;
    }

    public function getDataNascimento(): DateTime
    {
        return $this->dataNascimento;
    }

    public function setDataNascimento(DateTime $dataNascimento): void
    {
        $this->dataNascimento = $dataNascimento;
    }

    public function getSexo(): string
    {
        return $this->sexo;
    }

    public function setSexo(string $sexo): void
    {
        $this->sexo = $sexo;
    }

    public function getTelefone(): string
    {
        return $this->telefone;
    }

    public function setTelefone(string $telefone): void
    {
        $this->telefone = $telefone;
    }

    public function getEmail(): string
    {
        return $this->email;
    }

    public function setEmail(string $email): void
    {
        $this->email = $email;
    }

    public function getEndereco(): string
    {
        return $this->endereco;
    }

    public function setEndereco(string $endereco): void
    {
        $this->endereco = $endereco;
    }

    public function getRGM(): ?string
    {
        return $this->RGM;
    }

    public function setRGM(?string $RGM): void
    {
        $this->RGM = $RGM;
    }

    public function getExames(): array
    {
        return $this->exames;
    }

    public function setExames(array $exames): void
    {
        $this->exames = $exames;
    }

    public function addExame(Exame $exame): void
    {
        $this->exames[] = $exame;
    }
}


?>